### ISSUE TITLE
Generating JavaScript bundles failed - invalid argument, mkdir 

### ISSUE THREAD

**Issue by ibjorn** [graysonhicks/gatsby-plugin-loadable-components-ssr#25]
I recently upgraded to Gatsby v.3. I was keeping an eye on this plugin as I noticed the issues after upgrading. So today I decided to give it another bash with 3.0.0.

I run into this error, which seems to me like it could be a bug.

` ERROR #98123  WEBPACK

Generating JavaScript bundles failed

EINVAL: invalid argument, mkdir 'C:\Users\bjorn\www\blackalscoza\public\C:\Users\bjorn\www\blackalscoza\public' `

Things I've tried (admittedly, I'm a little inexperienced with the hydrate vs render functions, so a lot of this is trial and error for me):

- useHydrate: true and false
- My gatsby-browser.js:
  ` import loadable from '@loadable/component'

  export const replaceHydrateFunction = () => (element, container, callback) => {
      loadable.loadableReady(() => {
          ReactDOM.render(element, container, callback)
      })
  } `

I'm using theme-ui with the gatsby-plugin-theme-ui, as well as gatsby-plugin-transition-link which I suspect might be the culprits?

Without gatsby-plugin-loadable-components-ssr I don't get this error. As per the error, seems like there's an issue with how the 'public' folder gets created?

**Comment by kdichev** [comment]
some of our windows developers report the same issue, build works on macOS or Linux. As a workaround you can use WSL2 in the meantime until this is resolved.

**Comment by ibjorn** [comment]
Thanks for that. It's not too urgent at this stage, at least not moving development environment urgent. Will keep an eye out for an update.
Will do staging test on my VPS.

**Comment by kdichev** [comment]
on a side note do your project is in a `workspace` as ours is and I am seeing that there might be something wrong as the there is this folder containing the `loadable-stats-build-javascript.json` file when I execute `yarn worksapce dfdsdotcom build`

![image](https://user-images.githubusercontent.com/12297753/121602211-d310cd00-c9fb-11eb-86e7-de18ac69dee4.png)

**Comment by synim-sogody** [comment]
I've also recently faced this issue. @graysonhicks below you'll find a rough description of the issue as I understand it, two workarounds and a pull request #29 for a more permanent solution. Let me know if this helps in solving the issue at hand.

### Description

The issue seems like a classic absolute/relative path mix-up. While calling gatsby-node.js:20-27 and then emitting output assets on @loadable/webpack-plugin/lib/index.js:108,121 Windows treats this.opts.filename as a relative path and appends the project path resulting in the previous path error. 

@loadable/webpack-plugin/lib/index.js:102-125
```
      if (version === 4) {
        // webpack 4
        compiler.hooks.emit.tap(name, compilation => {
          const asset = this.handleEmit(compilation);

          if (asset) {
            compilation.assets[this.opts.filename] = asset;
          }
        });
      } else {
        // webpack 5
        compiler.hooks.make.tap(name, compilation => {
          compilation.hooks.processAssets.tap({
            name,
            stage: compiler.webpack.Compilation.PROCESS_ASSETS_STAGE_REPORT
          }, () => {
            const asset = this.handleEmit(compilation);

            if (asset) {
              compilation.emitAsset(this.opts.filename, asset);
            }
          });
        });
      }
```

### Workarounds

Seems this can be prevented if we set outputAsset to false on setwebpackConfig (gatsby-node.js:22-24) as follows.

gatsby-node.js:20-27
```
  if (stage === "build-javascript" || stage === "develop" || stage === "develop-html") {
    actions.setWebpackConfig({
      plugins: [new _webpackPlugin.default({
        filename: statsPath,
        writeToDisk: true,
        outputAsset: false
      })]
    });
  }
```

Another solution would be to set the filename directly to "loadable-stats-build-javascript.json" when on Windows.

gatsby-node.js:20-27
```
  if (stage === "build-javascript" || stage === "develop" || stage === "develop-html") {
    actions.setWebpackConfig({
      plugins: [new _webpackPlugin.default({
        filename: "loadable-stats-build-javascript.json",
        writeToDisk: true
      })]
    });
  }
```

### Solution

I've opened a pull request #29 for a more permanent solution, which sets this path dynamically based on platform OS.

**Comment by graysonhicks** [comment]
Thanks for the workarounds _and_ PR @synim-sogody !  I will get that released this week.

**Comment by graysonhicks** [comment]
Fix for this released with `3.2.0`
